src/components/menu/menu-segment-title/menu-segment-title.component.tsx

```diff
@@ -8,25 +8,32 @@ import tagComponent, {
 } from "../../../__internal__/utils/helpers/tags";
 
 export interface MenuTitleProps extends TagProps {
-  children: string;
+  children?: React.ReactNode;
+  text: string;
   /** Set the colour variant for a menuType */
   variant?: VariantType;
+  as?: "h2" | "h3" | "h4" | "h5" | "h6";
 }
 
 const MenuSegmentTitle = React.forwardRef<HTMLDivElement, MenuTitleProps>(
-  ({ children, variant = "default", ...rest }: MenuTitleProps, ref) => {
+  (
+    { children, variant = "default", text, as = "h2", ...rest }: MenuTitleProps,
+    ref
+  ) => {
     const menuContext = useContext(MenuContext);
 
     return (
       <StyledMenuItem inSubmenu>
         <StyledTitle
+          as={as}
```

Comment:
**question**: would it be worth filtering the `as` prop so only the supported heading level elements are accepted?

As its possible, despite this resulting in a TypeScript type error, that a JavaScript consumer could try to render the title as any element they like:

```tsx
<MenuSegmentTitle text="I am a title" as="div" /> 
``` 

since `as` is passed directly to the styled component `StyledTitle`

Reply:
I don't mind doing it but I'm of the opinion that if someone wants to explicitly ignore the types is kind of on them when it breaks etc

Reply:
I agree, though I wonder if some of our JavaScript-only consumers could accidentally do this without repercussions and flag an issue with `as` set to a non-heading element

Reply:
Ah that's a good point about prop-types 👍🏼 If we come across the same pattern elsewhere in the future, let's leave it in that case